robots/cmd/push-flakefinder-results/main.go

```diff
@@ -0,0 +1,145 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package main
+
+import (
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+)
+
+const (
+	dateRange24h  = "024h"
+	dateRange168h = "168h"
+	dateRange672h = "672h"
+)
+
+const (
+	testName = iota
+	testLane
+	severity
+	failed
+	succeeded
+	skipped
+)
+
+var dateRangeAllowedValues = map[string]struct{}{
+	dateRange24h:  {},
+	dateRange168h: {},
+	dateRange672h: {},
+}
+
+func main() {
+	var pushgatewayURL, dateRange, org, repo string
+	flag.StringVar(&pushgatewayURL, "pushgateway-url", "http://localhost:8080/", "pushgateway url to push values to")
+	flag.StringVar(&dateRange, "date-range", dateRange24h, "daterange of the report (one of 024h, 168h, 672h)")
+	flag.StringVar(&org, "github-org", "kubevirt", "github org")
+	flag.StringVar(&repo, "github-repo", "kubevirt", "github repo")
+	flag.Parse()
+
+	if _, exists := dateRangeAllowedValues[dateRange]; !exists {
+		log.Fatalf("Value %q not allowed for range, allowed values: %v", dateRange, dateRangeAllowedValues)
+	}
+
+	pusher := push.New(pushgatewayURL, fmt.Sprintf("flakefinder_results_%s", dateRange))
+
+	yesterday := time.Now().Add(-24 * time.Hour)
```

Comment:
Could `time.Now().Sub(24 * time.Hour)` be used here?

Reply:
IIRC for `time.Duration` there's only `Add`